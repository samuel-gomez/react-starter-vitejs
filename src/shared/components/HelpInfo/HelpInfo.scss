@import '@axa-fr/react-toolkit-core/dist/assets/scss/core.scss';

.af-popover__wrapper:has(.af-popover__container--short) {
  position: relative;
  display: block;
}

.af-popover__container--short {
  display: block;
  .af-popover__container-pop {
    word-break: break-word;
    width: 200px;
    text-align: center;
  }
}
