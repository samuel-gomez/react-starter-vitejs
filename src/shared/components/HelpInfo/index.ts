export { default } from './HelpInfo';
export type { THelpInfo } from './HelpInfo';
