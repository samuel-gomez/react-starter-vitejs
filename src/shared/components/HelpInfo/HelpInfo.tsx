import type { ReactNode } from 'react';
import { Popover } from '@axa-fr/react-toolkit-all';
import './HelpInfo.scss';

export type THelpInfo = {
  children: ReactNode;
  content?: ReactNode;
  isDisabled?: boolean;
  mode?: string;
  classModifier?: string;
};

const HelpInfo = ({ children, content, isDisabled, mode = 'hover', classModifier = 'short' }: THelpInfo) =>
  !isDisabled && content ? (
    <Popover mode={mode} classModifier={classModifier}>
      <Popover.Pop>{content}</Popover.Pop>
      <Popover.Over>{children}</Popover.Over>
    </Popover>
  ) : (
    <>{children}</>
  );

export default HelpInfo;
