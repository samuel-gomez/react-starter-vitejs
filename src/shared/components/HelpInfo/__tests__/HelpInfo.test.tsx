import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import HelpInfo from '..';

describe('<HelpInfo/>', () => {
  it('Should render HelpInfo', () => {
    const { asFragment } = render(
      <HelpInfo content={<p>info</p>}>
        <h1>Hello</h1>
      </HelpInfo>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render HelpInfo without popover when is disabled', () => {
    const { asFragment } = render(
      <HelpInfo content={<p>info</p>} isDisabled>
        <h1>Hello</h1>
      </HelpInfo>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render HelpInfo without popover when no content hover', () => {
    const { asFragment } = render(
      <HelpInfo>
        <h1>Hello</h1>
      </HelpInfo>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
