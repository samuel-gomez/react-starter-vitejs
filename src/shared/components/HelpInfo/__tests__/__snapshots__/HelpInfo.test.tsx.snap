// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HelpInfo/> > Should render HelpInfo 1`] = `
<DocumentFragment>
  <div
    class="af-popover__wrapper"
  >
    <div
      class="af-popover__container af-popover__container--short"
    >
      <div
        class="af-popover__container-over"
        role="presentation"
      >
        <h1>
          Hello
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<HelpInfo/> > Should render HelpInfo without popover when is disabled 1`] = `
<DocumentFragment>
  <h1>
    Hello
  </h1>
</DocumentFragment>
`;

exports[`<HelpInfo/> > Should render HelpInfo without popover when no content hover 1`] = `
<DocumentFragment>
  <h1>
    Hello
  </h1>
</DocumentFragment>
`;
